lib/auth/status.ex

```diff
@@ -15,27 +16,31 @@ defmodule Auth.Status do
   @doc false
   def changeset(status, attrs) do
     status
-    |> cast(attrs, [:text])
+    |> cast(attrs, [:text, :desc])
     |> validate_required([:text])
   end
 
-  def create_status(text, person) do
+  def create_status(attrs, person) do
     %Status{}
-    |> changeset(%{text: text})
+    |> changeset(attrs)
     |> put_assoc(:person, person)
     |> Repo.insert!()
   end
 
-  def upsert_status(text) do
-    case Auth.Repo.get_by(__MODULE__, text: text) do
+  def upsert_status(attrs) do
+    case Auth.Repo.get_by(__MODULE__, text: Map.get(attrs, "text")) do
       # create status
       nil ->
         email = System.get_env("ADMIN_EMAIL")
```

Comment:
Personally, I'd set a `@admin_email` module attribute at the top of the module instead of fetching from env each time.

E.g. 

```elixir
@admin_email System.get_env("ADMIN_EMAIL")
```

Reply:
Totally agree. Dunno why I did that. 🤦  (_already updated it and simplified the code_) ✅